Firestore/core/src/firebase/firestore/local/local_documents_view.mm

```diff
@@ -215,6 +217,38 @@
   return results;
 }
 
+// It is possible that a `PatchMutation` can make a document match a query, even
+// if the version in the `RemoteDocumentCache` is not a match yet (waiting for
+// server to ack). To handle this, we find all document keys affected by the
+// `PatchMutation`s that are not in `existingDocs` yet, and back fill them via
+// `remoteDocumentCache.getAll`, otherwise those `PatchMutation`s will be
+// ignored because no base document can be found, and lead to missing results
+// for the query.
+DocumentMap LocalDocumentsView::AddMissingBaseDocument(
+    const std::vector<FSTMutationBatch*>& matchingBatches,
+    const DocumentMap& existingDocs) {
```

Comment:
This function's signature mandates an unnecessary copy. Even though `DocumentMap` is relatively cheap to copy, it strikes me as an avoidable inefficiency. I'd suggest moving the map instead. Make the function take `existing_docs` by value:
```cpp
DocumentMap existing_docs) {
```
and then move `results` into the function parameter at the call point:
```cpp
results = AddMissingBaseDocuments(matching_batches, std::move(results));
 ```

Reply:
Done.